src/axes/categoryAxis.ts

```diff
@@ -211,6 +293,53 @@ namespace Plottable.Axes {
       return this._maxLabelTickLength() + this.tickLabelPadding();
     }
 
+    private _calcTextPadding(): { translate: { x: number, y: number }, pad: { x: number, y: number }} {
+      // Padding always moves the label *away* from the outer edge of
+      // the axis (tickLabelPadding moves labels away from the inner
+      // edge of the axis already).
+
+      // The pad value gives the amount of padding that must be
+      // subtracted from overall space when renderign text (import for
+      // proper line breaking).
+      //
+      // The translate value gives the amount the label should be
+      // moved to give the proper padding.
+      //
+      // Pad will always be a positive amount, but translate can be
+      // negative depending on rotation of labels and axis
+      // orientation.
+      let pad = { x: 0, y: 0 };
+      let translate = { x: 0, y: 0 };
+
+      if (this.tickLabelAngle() === 0 && ! this.isHorizontal()) {
+        if (this.orientation() === "right" && this.tickTextAlignment() === "right") {
+          pad.x = this.tickTextPadding();
+          translate.x = this.tickTextPadding() * -1;
+        } else if (this.orientation() === "left" && this.tickTextAlignment() === "left") {
+          pad.x = this.tickTextPadding();
+          translate.x = this.tickTextPadding();
+        }
+      } else if (this.tickLabelAngle() === -90) {
+        if (this.orientation() === "top" && this.tickTextAlignment() === "right") {
+          pad.y = this.tickTextPadding();
+          translate.y = this.tickTextPadding();
+        } else if (this.orientation() === "bottom" && this.tickTextAlignment() === "left") {
+          pad.y = this.tickTextPadding();
+          translate.y = this.tickTextPadding() * -1;
+        }
+      } else if (this.tickLabelAngle() === 90) {
+        if (this.orientation() === "top" && this.tickTextAlignment() === "left") {
+          pad.y = this.tickTextPadding();
+          translate.y = this.tickTextPadding();
+        } else if (this.orientation() === "bottom" && this.tickTextAlignment() === "right") {
+          pad.y = this.tickTextPadding();
+          translate.y = this.tickTextPadding() * -1;
+        }
+      }
+
+      return { pad: pad, translate: translate };
```

Comment:
why not just return translate and just make the space calculation take into account the possibility of negative values?